modules/analysis-common/src/main/java/org/opensearch/analysis/common/CommonAnalysisModulePlugin.java

```diff
@@ -191,9 +196,16 @@ public Collection<Object> createComponents(
         Supplier<RepositoriesService> repositoriesServiceSupplier
     ) {
         this.scriptService.set(scriptService);
+        try {
+            this.analysisModule = new AnalysisModule(environment,
+                List.of(this));
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
```

Comment:
I'm a little concerned with this approach of creating a new `AnalysisModule` with its own `AnalysisRegistry`.

If you define a custom analyzer, are you able to reference it from the `synonym_analyzer` parameter? I could be mistaken, but I think this module's registry only knows the built-in analyzers.

Reply:
One thing that we could do is add a default method `getTokenFilters(AnalysisModule)` to the `AnalysisPlugin` interface that calls the existing `getTokenFilters()`. Then we could override the new method in `CommonAnalysisModulePlugin`. You'd also need to call the new method from `setupTokenFilters`.

It does mean that we would pass a reference to the partially-constructed `AnalysisModule` to `getTokenFilters`. As long as a consumer only holds onto the reference and doesn't try doing anything with it right away, it should be okay.

Reply:
Thanks @msfroh, 

- With creation of new `AnalysisModule`, I was able to reference it from the `synonym_analyzer` parameter, also in my test not only the built-in analyzers but custom analyzer also worked https://github.com/opensearch-project/OpenSearch/issues/16263#issuecomment-2440431429.

- I agree with your suggestion we can re-use the `AnalysisModule` and dont have to instantiate a new one. In the new commit I have updated the `AnalysisPlugin` section where `getTokenFilters` takes `AnalysisModule` as input. I have also updated the `CommonAnalysisModulePlugin` and `setupTokenFilters`.

Reply:
Also I will revert the changes made to `buildSrc/src/main/java/org/opensearch/gradle/testclusters/OpenSearchNode.java`, I was trying to test by adding `config/hunspell/en_US/en-US.aff` to config folder before startup (`./gradlew run -PnumNodes=3) but `OpenSearchNode.java` does not copy the folder files, so was  experimenting few things with `OpenSearchNode.java` I will open an issue separately for this topic.